### PR TITLE
fix: Toolbar z-Index, trailing paragraphs

### PR DESCRIPTION
**Pull request by fsbraun**
## Summary by Sourcery

Adjust toolbar positioning and editor behavior to fix z-index issues and avoid unwanted trailing paragraphs in inline and fixed editors.

Bug Fixes:
- Prevent the editor schema from appending an empty trailing paragraph after headings.
- Ensure the Tiptap toolbar remains visible with correct stacking order and positioning in both inline and fixed modes.
- Handle inline rich text wrappers more robustly, including semantic container tags and optional .cms-content-start markers.

Enhancements:
- Extend inline editor wrapper detection to support additional semantic container elements.
- Add scroll-aware toolbar pinning logic that works around overflow constraints by manually adjusting absolute positioning.

Tests:
- Add unit tests for inline rich text wrapper initialization, including wrapper tags and .cms-content-start handling.
- Add document schema tests to assert no trailing empty paragraphs are added after headings.

### CODE REVIEW

File: private/js/tiptap_plugins/cms.toolbar.js [hunk 1]
```diff
@@ -330,6 +325,76 @@ function _createTopToolbarPlugin(editor, filter) {
...
+function _pinToolbarOnScroll(editorElement, toolbar, isInline) {
+    const cmsToolbarHeight = parseInt(
+        getComputedStyle(document.documentElement)
+            .getPropertyValue('--cms-toolbar-height') || '0', 10
+    );
+
+    function update() {
+        const rect = editorElement.getBoundingClientRect();
+
+        if (isInline) {
```
[the remark this answers is not in the document]
Comment (fsbraun) [reply private/js/tiptap_plugins/cms.toolbar.js:365]: @sourcery-ai Now the scroll listener is destroyed when the editor is destroyed
